tests/test_basics.py

```diff
@@ -91,7 +91,65 @@ def test_event_id(sentry_init, capture_events):
     assert Hub.current.last_event_id() == event_id
 
 
-def test_option_callback(sentry_init, capture_events, monkeypatch):
+def test_option_before_send(sentry_init, capture_events, monkeypatch):
+    def before_send(event, hint):
+        event["extra"] = {"before_send_called": True}
+        return event
+
+    def do_this():
+        try:
+            raise ValueError("aha!")
+        except Exception:
+            capture_exception()
+
+    sentry_init(before_send=before_send)
+    events = capture_events()
+
+    do_this()
+
+    (event,) = events
+    assert event["extra"] == {"before_send_called": True}
+
+    sentry_init()
+    events = capture_events()
+
+    do_this()
+
+    (event,) = events
+    assert "before_send_called" not in event["extra"]
+
+
+def test_option_before_send_transaction(sentry_init, capture_events, monkeypatch):
+    def before_send_transaction(event, hint):
+        assert event["type"] == "transaction"
+        event["extra"] = {"before_send_transaction_called": True}
+        return event
+
+    sentry_init(
```

Comment:
calling init multiple times in a test is weird, can you use `pytest.mark.parametrize` for the different combinations

Reply:
Removed the second init as a whole, because it is just testing normal sdk behaviour that is tested somewhere else.